ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client/Jumping.hs

```diff
@@ -738,13 +753,58 @@ unregisterClient ::
   PeerContext m peer blk ->
   STM m ()
 unregisterClient context = do
-  modifyTVar (handlesVar context) $ Map.delete (peer context)
+  cschcRemoveHandle (handlesCol context) (peer context)
   let context' = stripContext context
   readTVar (cschJumping (handle context)) >>= \case
     Disengaged{} -> pure ()
     Jumper{} -> pure ()
     Objector{} -> electNewObjector context'
-    Dynamo{} -> electNewDynamo context'
+    Dynamo{} -> void $ electNewDynamo context'
+
+-- | Elects a new dynamo by demoting the given dynamo to a jumper, moving the
+-- peer to the end of the queue of chain sync handles and electing a new dynamo.
```

Comment:
Because of `promoteToDynamo`, it also demotes the objector and the unhappy jumpers back to happy jumpers, right?

Reply:
That't true.

Reply:
Done in 72a5077edd144fb9aa7581cf10c50e3fa253679a.